### PR TITLE
[Driver][SYCL] Adjust /MD settings for SYCL device compilations

### PR DESCRIPTION
**Pull request by mdtoguchi**
When setting /MD (by default) we do not want to pass the _MT and _DLL
predefines to the device compilations, only the host for clang-cl

### CODE REVIEW

**Review by v-klochkov — APPROVED** [review]

**Comment by mdtoguchi** [comment]
I was a little over zealous in determining device compilations - cleaned that up.  Also make sure that the error message of usage is only emitted once.

**Review by v-klochkov — APPROVED** [review]

File: clang/test/Driver/sycl-MD-default.cpp [hunk 1]
```diff
@@ -6,6 +6,7 @@
 // RUN:   | FileCheck -check-prefix=CHK-DEFAULT %s
 // RUN: %clang_cl -### -MDd -fsycl -c %s 2>&1 \
 // RUN:   | FileCheck -check-prefix=CHK-DEFAULT %s
+// CHK-DEFAULT-NOT: "-fsycl-is-device" {{.*}} "-D_MT" "-D_DLL"
 // CHK-DEFAULT: "-D_MT" "-D_DLL"
```
Comment (AGindinson) [inline clang/test/Driver/sycl-MD-default.cpp:10]: For clarity & better coherence with the new check:
```suggestion
// CHK-DEFAULT: "-fsycl-is-host" {{.*}}  "-D_MT" "-D_DLL"
```

**Review by AGindinson — APPROVED** [review]
LGTM. I have to note that the mask/lambda solution feels really neat to me :)

**Review by v-klochkov — APPROVED** [review]

**Comment by v-klochkov** [comment]
Oops, I did not notice sycl-MD-default.cpp test failed.
@mdtoguchi - would you please fix it?

**Review by AGindinson — APPROVED** [review]

**Pull request merged**